nifi-nar-bundles/nifi-jms-bundle/nifi-jms-processors/src/main/java/org/apache/nifi/jms/processors/ConsumeJMS.java

```diff
@@ -173,6 +175,23 @@
         relationships = Collections.unmodifiableSet(_relationships);
     }
 
+    private static boolean isDurableSubscriber(final ProcessContext context) {
+        final Boolean durableBoolean = context.getProperty(DURABLE_SUBSCRIBER).evaluateAttributeExpressions().asBoolean();
+        return durableBoolean == null ? false : durableBoolean;
+    }
+
+    private static boolean isShared(final ProcessContext context) {
+        final Boolean sharedBoolean = context.getProperty(SHARED_SUBSCRIBER).evaluateAttributeExpressions().asBoolean();
+        return sharedBoolean == null ? false : sharedBoolean;
+    }
+
+    @OnScheduled
+    public void onSchedule(ProcessContext context) {
```

Comment:
I believe you have a typo here. I think the logic should be `context.getMaxConcurrentTasks() > 1 && isDurableSubscriber(context) && !isShared(context)`

Can you please add a unit test verifying this new behavior? Also, is it possible to add unit test(s) to verify that the issue you're trying to address (changing client IDs for durable non-shared clients on processor restart) is resolved?

Reply:
You are right, thank you. I will push the fix.

Reply:
Fixed and added unit tests to verify the behavior. You can check without this fix, the unit test will miss a message.